### ISSUE TITLE
Design Size Maximum Flow per Zone Floor Area during Reheat overflows if zone has zero floor area

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5273]
This file with no floor in Zone1 reports ***\* in a release build and fails in a debug build.

[5ZoneVAV-ChilledWaterStorage-Mixed-MoreReporting-DeleteZone1Floor-Crash.idf.txt](https://github.com/NREL/EnergyPlus/files/16392/5ZoneVAV-ChilledWaterStorage-Mixed-MoreReporting-DeleteZone1Floor-Crash.idf.txt)

`Component Sizing Information, AirTerminal:SingleDuct:VAV:Reheat, ZN_1_FLR_1_SEC_1 VAV BOX COMPONENT, Design Size Maximum Flow per Zone Floor Area during Reheat [m3/s-m2], *****`

**Comment by rraustad** [comment]
So is 0 floor area valid?

```
// Set up Floor Areas for Zones
if ( ! SurfError ) {
    for ( ZoneNum = 1; ZoneNum <= NumOfZones; ++ZoneNum ) {
        for ( SurfNum = Zone( ZoneNum ).SurfaceFirst; SurfNum <= Zone( ZoneNum ).SurfaceLast; ++SurfNum ) {
            if ( Surface( SurfNum ).Class == SurfaceClass_Floor ) {
                Zone( ZoneNum ).FloorArea += Surface( SurfNum ).Area;
                Zone( ZoneNum ).HasFloor = true;
            }
            if ( Surface( SurfNum ).Class == SurfaceClass_Roof ) {
                Zone( ZoneNum ).HasRoof = true;
            }
        }
    }
    ErrCount = 0;
    for ( ZoneNum = 1; ZoneNum <= NumOfZones; ++ZoneNum ) {
        Zone( ZoneNum ).CalcFloorArea = Zone( ZoneNum ).FloorArea;
        if ( Zone( ZoneNum ).UserEnteredFloorArea != AutoCalculate ) {
            // Check entered vs calculated
            if ( Zone( ZoneNum ).UserEnteredFloorArea > 0.0 ) { // User entered zone floor area,
                // produce message if not near calculated
                if ( Zone( ZoneNum ).CalcFloorArea > 0.0 ) {
                    diffp = std::abs( Zone( ZoneNum ).CalcFloorArea - Zone( ZoneNum ).UserEnteredFloorArea ) / Zone( ZoneNum ).UserEnteredFloorArea;
                    if ( diffp > 0.05 ) {
                        ++ErrCount;
                        if ( ErrCount == 1 && ! DisplayExtraWarnings ) {
                            ShowWarningError( RoutineName + "Entered Zone Floor Areas differ from calculated Zone Floor Area(s)." );
                            ShowContinueError( "...use Output:Diagnostics,DisplayExtraWarnings; to show more details on individual zones." );
                        }
                        if ( DisplayExtraWarnings ) {
                            // Warn user of using specified Zone Floor Area
                            ShowWarningError( RoutineName + "Entered Floor Area entered for Zone=\"" + Zone( ZoneNum ).Name + "\" significantly different from calculated Floor Area" );
                            ShowContinueError( "Entered Zone Floor Area value=" + RoundSigDigits( Zone( ZoneNum ).UserEnteredFloorArea, 2 ) + ", Calculated Zone Floor Area value=" + RoundSigDigits( Zone( ZoneNum ).CalcFloorArea, 2 ) + ", entered Floor Area will be used in calculations." );
                        }
                    }
                }
                Zone( ZoneNum ).FloorArea = Zone( ZoneNum ).UserEnteredFloorArea;
                Zone( ZoneNum ).HasFloor = true;
            }
        } else {
            Zone( ZoneNum ).FloorArea = Zone( ZoneNum ).CalcFloorArea; // redundant, already done.
        }
    }
}
```

**Comment by rraustad** [comment]
Zone.HasFloor used here:

```
void
CalculateZoneVolume(
    bool & ErrorsFound, // If errors found in input
    Array1S_bool const CeilingHeightEntered
)
{

        if ( ! Zone( ZoneNum ).HasFloor ) {
            ShowWarningError( "No floor exists in Zone=\"" + Zone( ZoneNum ).Name + "\", zone floor area is zero. All values for this zone that are entered per floor area will be zero." );
        }
```

**Comment by rraustad** [comment]
So this implies to catch this in the SingleDuct model:

```
 ** Warning ** No floor exists in Zone="ZN_1_FLR_1_SEC_1", zone floor area is zero. All values for this zone that are entered per floor area will be zero.
```

...here:

```
if ( IsAutoSize ) {
    Sys( SysNum ).MaxAirVolFlowRateDuringReheat = MaxAirVolFlowRateDuringReheatDes;
    ReportSizingOutput( Sys( SysNum ).SysType, Sys( SysNum ).SysName, "Design Size Maximum Flow per Zone Floor Area during Reheat [m3/s-m2]", MaxAirVolFlowRateDuringReheatDes / Sys( SysNum ).ZoneFloorArea );
} else { // Hard size with sizing data
```

**Comment by rraustad** [comment]
This:

```
if ( IsAutoSize ) {
    if( Sys( SysNum ).ZoneFloorArea > 0.0 ) {
        Sys( SysNum ).MaxAirVolFlowRateDuringReheat = MaxAirVolFlowRateDuringReheatDes;
        ReportSizingOutput( Sys( SysNum ).SysType, Sys( SysNum ).SysName, "Design Size Maximum Flow per Zone Floor Area during Reheat [m3/s-m2]", MaxAirVolFlowRateDuringReheatDes / Sys( SysNum ).ZoneFloorArea );
    } else {
        Sys( SysNum ).MaxAirVolFlowRateDuringReheat = 0.0;
        Sys( SysNum ).MaxAirVolFractionDuringReheat = 0.0;
        ReportSizingOutput( Sys( SysNum ).SysType, Sys( SysNum ).SysName, "Design Size Maximum Flow per Zone Floor Area during Reheat [m3/s-m2]", 0.0 );
    }
} else { // Hard size with sizing data
```

yields this:

```
Component Sizing Information, AirTerminal:SingleDuct:VAV:Reheat, ZN_1_FLR_1_SEC_1 VAV BOX COMPONENT, Design Size Maximum Air Flow Rate [m3/s], 1.11327
Component Sizing Information, AirTerminal:SingleDuct:VAV:Reheat, ZN_1_FLR_1_SEC_1 VAV BOX COMPONENT, Design Size Maximum Flow per Zone Floor Area during Reheat [m3/s-m2], 0.00000
Component Sizing Information, AirTerminal:SingleDuct:VAV:Reheat, ZN_1_FLR_1_SEC_1 VAV BOX COMPONENT, Design Size Maximum Reheat Water Flow Rate [m3/s], 6.10158E-004
Component Sizing Information, AirTerminal:SingleDuct:VAV:Reheat, ZN_1_FLR_1_SEC_1 VAV BOX COMPONENT, Design Size Reheat Coil Sizing Air Volume Flow Rate [m3/s], 0.90751
Component Sizing Information, AirTerminal:SingleDuct:VAV:Reheat, ZN_1_FLR_1_SEC_1 VAV BOX COMPONENT, Design Size Reheat Coil Sizing Inlet Air Temperature [C], 16.70000
Component Sizing Information, AirTerminal:SingleDuct:VAV:Reheat, ZN_1_FLR_1_SEC_1 VAV BOX COMPONENT, Design Size Reheat Coil Sizing Inlet Air Humidity Ratio [kgWater/kgDryAir], 8.00000E-003
```

**Comment by Myoldmopar** [comment]
Closed via #5568
